### PR TITLE
Extend logprob inference for scans with carried auxiliary states

### PR DESCRIPTION
**Pull request by ricardoV94**
PyMC can now infer graphs like MA(2)

```python
import numpy as np
import pytensor

import pymc as pm

init_eps = pm.DiracDelta.dist(np.zeros(2, dtype="float64"))
rho = [0.3, 0.9]
sigma = 1
n_steps = 100

def ma2_step(eps_tm2, eps_tm1, rho, sigma):
    mu = eps_tm1 * rho[0] + eps_tm2 * rho[1]
    y = pm.Normal.dist(mu, sigma)
    eps = y - mu
    return eps, y

[_, ma2], _ = pytensor.scan(
    fn=ma2_step,
    outputs_info=[{"initial": init_eps, "taps": range(-2, 0)}, None],
    non_sequences=[rho, sigma],
    n_steps=n_steps,
    name="ma2",
)

ma2_test = ma2.eval()
pm.logp(ma2, ma2_test).sum().eval()
```

CC @jessegrabowski

### CODE REVIEW

File: pymc/testing.py [hunk 1]
```diff
@@ -958,5 +959,18 @@ def seeded_numpy_distribution_builder(dist_name: str) -> Callable:
...
+        if owner:
+            inputs = list(reversed(owner.inputs))
+
+            if isinstance(owner.op, HasInnerGraph):
+                inputs += owner.op.inner_outputs
+
+            return inputs
+
+    for v in walk([var], expand, False):
+        if v.owner and isinstance(v.owner.op, (RandomVariable, MeasurableVariable)):
```
Comment (larryshamalama) [inline pymc/testing.py:975]: Are `RandomVariable`s not measurable?

File: pymc/testing.py [hunk 2]
```diff
@@ -958,5 +959,18 @@ def seeded_numpy_distribution_builder(dist_name: str) -> Callable:
 
 
 def assert_no_rvs(var):
```
Comment (larryshamalama) [inline pymc/testing.py:961]: IIUC, would the name `assert_no_measurable_rvs` be more fitting?

**Review by larryshamalama — COMMENTED** [review]
Ouf, `Scan`s a bit beyond my area of comfort. I just have minor comments for this one

File: pymc/testing.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/testing.py:975]: They are, but this makes the code more readable I think. Most people don't know or need to know about MeasurableVariables so I rather make the function name reflect RVs and explicitly include them in the isinstance check

File: pymc/testing.py (hunk 2 above)
Comment (ricardoV94) [reply pymc/testing.py:961]: See reasoning above

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Need to add tags so this shows up in release notes.

**Comment by ricardoV94** [comment]
> Need to add tags so this shows up in release notes.

Already had. Without tags it still shows up in the release notes, but in the catch-all "maintenance" bucket. The only case it doesn't show up is if it has the label "no releasenotes"

File: pymc/testing.py (hunk 1 above)
Comment (larryshamalama) [reply pymc/testing.py:975]: Okay, I understand

**Pull request merged**